### PR TITLE
94 implement support for configuration and creation of beans via methods

### LINKED ISSUE #94: Implement support for @Configuration and creation of beans via methods

### PR DESCRIPTION
**Pull request by mykolaFilimonov**

### CODE REVIEW

**Review by OlehYakovenko — APPROVED** [review]

File: src/main/java/io/github/bobocodebreskul/context/annotations/BringBean.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package io.github.bobocodebreskul.context.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface BringBean {
```
Comment (JoshukLite) [inline src/main/java/io/github/bobocodebreskul/context/annotations/BringBean.java:29]: No documentation. 
Good to have usage example.

File: src/main/java/io/github/bobocodebreskul/context/annotations/BringConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+//TODO: write doc
+
+@BringComponent
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface BringConfiguration {
```
Comment (JoshukLite) [inline src/main/java/io/github/bobocodebreskul/context/annotations/BringConfiguration.java:31]: Docs with example.

File: src/main/java/io/github/bobocodebreskul/context/annotations/BringBean.java (hunk 1 above)
Comment (mykolaFilimonov) [reply src/main/java/io/github/bobocodebreskul/context/annotations/BringBean.java:29]: added

File: src/main/java/io/github/bobocodebreskul/context/annotations/BringConfiguration.java (hunk 2 above)
Comment (mykolaFilimonov) [reply src/main/java/io/github/bobocodebreskul/context/annotations/BringConfiguration.java:31]: added

File: src/main/java/io/github/bobocodebreskul/context/registry/BeanDefinitionReader.java [hunk 3]
```diff
@@ -139,4 +144,46 @@ private static <T> String getBeanDefinitionScope(Class<T> beanClass, String bean
...
+  }
+
+  private ConfigurationBeanDefinition mapBeanMethodToBeanDefinition(Method beanMethod) {
+
+    log.debug("mapBeanMethodToBeanDefinition method invoked: beanClass={}",
+        beanMethod.getReturnType());
+    log.info("Registering the bean definition of class {}", beanMethod.getReturnType().getName());
+    Class<?> beanType = beanMethod.getReturnType();
+    Object configClassInstance = instantiateConfigurationClass(beanMethod.getDeclaringClass());
+
```
Comment (AlexandrKarpachov) [inline src/main/java/io/github/bobocodebreskul/context/registry/BeanDefinitionReader.java:162]: I think we need to notice in the documentation that we don't support Autowiring for configuration classes. Cause it looks like a bug. Or fix it if we will have enough time.

**Review by AlexandrKarpachov — APPROVED** [review]

**Pull request merged**